core/src/test/java/io/grpc/util/interceptor/server/UtilServerInterceptorsTest.java

```diff
@@ -51,42 +48,27 @@
  */
 @RunWith(JUnit4.class)
 public class UtilServerInterceptorsTest {
-  @Mock
-  private Marshaller<String> requestMarshaller;
-
-  @Mock
-  private Marshaller<Integer> responseMarshaller;
+  private MethodDescriptor<String, Integer> flowMethod = TestMethodDescriptors.noopMethod();
+  private ServerCall<String, Integer> call = Mockito.spy(new NoopServerCall<String, Integer>());
+  private final Metadata headers = new Metadata();
 
-  @Mock
+  private ServerServiceDefinition serviceDefinition;
   private ServerCallHandler<String, Integer> handler;
-
-  @Mock
   private ServerCall.Listener<String> listener;
 
-  private MethodDescriptor<String, Integer> flowMethod;
-
-  private ServerCall<String, Integer> call = new NoopServerCall<String, Integer>();
-
-  private ServerServiceDefinition serviceDefinition;
-
-  private final Metadata headers = new Metadata();
-
   /** Set up for test. */
   @Before
   public void setUp() {
     MockitoAnnotations.initMocks(this);
-    flowMethod = MethodDescriptor.<String, Integer>newBuilder()
-        .setType(MethodType.UNKNOWN)
-        .setFullMethodName("basic/flow")
-        .setRequestMarshaller(requestMarshaller)
-        .setResponseMarshaller(responseMarshaller)
-        .build();
-
-    Mockito.when(handler.startCall(
-        Mockito.<ServerCall<String, Integer>>any(), Mockito.<Metadata>any()))
-        .thenReturn(listener);
-
-    serviceDefinition = ServerServiceDefinition.builder(new ServiceDescriptor("basic", flowMethod))
+    handler = new ServerCallHandler<String, Integer>() {
```

Comment:
This initialization can be done at declaration time, and then serviceDefinition can also be initialized when declared. That would leave just initMocks in setup.